### PR TITLE
Add city42 keyboard

### PR DESCRIPTION
**Pull request by controller-works**
## Description

Adds support for city42 keyboard from Controller Works. It is a unibody split keyboard based on the RP2040 with a Cirque Trackpad.

## QMK Pull Request 

[Link to QMK Pull Request](https://github.com/qmk/qmk_firmware/pull/21727)

## Checklist

- [x] The VIA support for this keyboard is **MERGED** in QMK master already **(MANDATORY)**
- [x] The VIA definition follows the guide here: https://caniusevia.com/docs/layouts
- [x] I have a V3 JSON version for this keyboard definition.**(MANDATORY)**
- [x] I have tested this keyboard definition using VIA's "Design" tab.
- [x] I have tested this keyboard definition with firmware on a device.
- [x] I have assigned alpha keys and modifier keys with the correct colors.
- [x] The Vendor ID is not `0xFEED`

### CODE REVIEW

File: v3/controllerworks/city42/city42.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "Controller Works city42",
+  "vendorId": "0x4357",
+  "productId": "0x0005",
+  "matrix": {"rows": 4, "cols": 12},
+  "menus": [ "qmk_rgb_matrix" ],
```
Comment (Xelus22) [inline v3/controllerworks/city42/city42.json:6]: Looks like in QMK not all effect are enabled. Please align QMK with VIA or VIA with QMK source

**Review by Xelus22 — CHANGES_REQUESTED** [review]
Please remove `src/` folder definiton, which is the old V2 definitions which are legacy. 
Your QMK Firmware was only merged 3 weeks ago.

File: v3/controllerworks/city42/city42.json (hunk 1 above)
Comment (controller-works) [reply v3/controllerworks/city42/city42.json:6]: I have added all of the animations to QMK in this PR (not yet approved):
[Add all RGB Matrix Animations to city42](https://github.com/qmk/qmk_firmware/pull/21992)

Comment (controller-works) [reply v3/controllerworks/city42/city42.json:6]: The PR above with all of the RGB matrix animations was merged.

**Review by Xelus22 — APPROVED** [review]

**Pull request merged**